src/ui/SmartSnippet/SmartSnippetCollapsibleSuggestion.ts

```diff
@@ -0,0 +1,112 @@
+import { IPartialQuestionAnswerResponse } from '../../rest/QuestionAnswerResponse';
+import { uniqueId } from 'underscore';
+import { AccessibleButton } from '../../utils/AccessibleButton';
+import { SVGIcons } from '../../utils/SVGIcons';
+import { attachShadow } from '../../misc/AttachShadowPolyfill';
+import { $$, Dom } from '../../utils/Dom';
+import { l } from '../../strings/Strings';
+
+const QUESTION_CLASSNAME = `coveo-smart-snippet-suggestions-question`;
+const QUESTION_TITLE_CLASSNAME = `${QUESTION_CLASSNAME}-title`;
+const QUESTION_TITLE_LABEL_CLASSNAME = `${QUESTION_TITLE_CLASSNAME}-label`;
+const QUESTION_TITLE_CHECKBOX_CLASSNAME = `${QUESTION_TITLE_CLASSNAME}-checkbox`;
+const QUESTION_SNIPPET_CLASSNAME = `${QUESTION_CLASSNAME}-snippet`;
+const QUESTION_SNIPPET_HIDDEN_CLASSNAME = `${QUESTION_SNIPPET_CLASSNAME}-hidden`;
+const SHADOW_CLASSNAME = `${QUESTION_SNIPPET_CLASSNAME}-content`;
+
+export class SmartSnippetCollapsibleSuggestion {
+  private readonly labelId = uniqueId(QUESTION_TITLE_LABEL_CLASSNAME);
+  private readonly snippetId = uniqueId(QUESTION_SNIPPET_CLASSNAME);
+  private readonly checkboxId = uniqueId(QUESTION_TITLE_CHECKBOX_CLASSNAME);
+  private contentLoaded: Promise<void>;
+  private collapsibleContainer: Dom;
+  private checkbox: Dom;
+  private expanded = false;
+
+  constructor(private readonly questionAnswer: IPartialQuestionAnswerResponse, private readonly innerCSS?: string) {}
+
+  public get loading() {
+    return this.contentLoaded;
+  }
+
+  public build() {
+    const collapsibleContainer = this.buildCollapsibleContainer(
+      this.questionAnswer.answerSnippet,
+      this.questionAnswer.question,
+      this.innerCSS && this.buildStyle(this.innerCSS)
+    );
+    const title = this.buildTitle(this.questionAnswer.question);
+    this.updateExpanded();
+    return $$(
+      'li',
+      {
+        className: QUESTION_CLASSNAME,
+        ariaLabelledby: this.labelId
+      },
+      title,
+      collapsibleContainer
+    ).el as HTMLLIElement;
+  }
+
+  private buildStyle(innerCSS: string) {
+    const styleTag = document.createElement('style');
+    styleTag.innerHTML = innerCSS;
+    return styleTag;
+  }
+
+  private buildTitle(question: string) {
+    const checkbox = this.buildCheckbox(question);
+    const label = $$('span', { className: QUESTION_TITLE_LABEL_CLASSNAME, id: this.labelId });
+    label.text(question);
```

Comment:
I think the question could be used to inject XSS, since it's being rendered inside the `title` as html. Something to check and possibly add a UT for.

Reply:
`label.text()` sets the `innerText` or `textContent` of an element, which is escaped. The answer is a different story, though...